pets/etcd/etcd-petset.yaml

```diff
@@ -0,0 +1,121 @@
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+  name: etcd
+  labels:
+    app: etcd
+spec:
+  ports:
+  - port: 2380
+    name: peer
+  - port: 2379
+    name: client
+  clusterIP: None
+  selector:
+    app: etcd
+---
+apiVersion: apps/v1alpha1
+kind: PetSet
+metadata:
+  name: etcd
+spec:
+  serviceName: "etcd"
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: etcd
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+    spec:
+      containers:
+      - name: etcd
+        image: gcr.io/google_containers/etcd-amd64:2.2.5
+        ports:
+        - containerPort: 2380
+          name: peer
+        - containerPort: 2379
+          name: client
+        env:
+        - name: CLUSTER_SIZE
```

Comment:
Reading the scaling workflow [1], there could be a way. As pets are added/remove in deterministic order, one could:

1. rename ``CLUSTER_SIZE`` into ``INITIAL_CLUSTER_SIZE``
1. bootstrap the etcd cluster
1. if a new pet is created, the next pet has ``NAME-${INITIAL_CLUSTER_SIZE}`` name so based on that I can decide if the new pet is part of bootstrapping phase or not (for a new pet after bootstrapping, ``$ID >= $INITIAL_CLUSTER_SIZE``, where ``NAME-${ID}`` is the unique pet name, always holds).
1. the new pet is added via the runtime configuration process. If the pet dies, it follows the same process. Once the pet is running and ready, next pet (if there is any) is added the same way.
1. assuming there is a way to notify the pod about its removal, ``etcdctl member remove`` is run.

[1] https://github.com/kubernetes/kubernetes.github.io/blob/release-1.3/docs/user-guide/petset.md#scaling-a-pet-set